test/test_ticks.py

```diff
@@ -0,0 +1,27 @@
+from ipctest import IpcTest
+import i3ipc
+
+class TestTicks(IpcTest):
+    events = []
+
+    def on_tick(self, i3, e):
+        self.events.append(e)
+        if len(self.events) == 3:
+            i3.main_quit()
+
+    def test_tick_event(self, i3):
+        i3.on('tick', self.on_tick)
+        i3.event_socket_setup()
+        i3.send_tick()
+        i3.send_tick('hello world')
+        while not i3.event_socket_poll():
```

Comment:
This hangs forever if ticks are not received.

Reply:
I guess there is no other way than copying the whole `.main()`?

Reply:
I put a `timeout` param in `main` for just this purpose. Maybe that should be pushed up into the `event_socket_poll` function?

Reply:
Not a bad idea, if optional.